### PR TITLE
Feature/gitpod decorator migration

### PR DESCRIPTION
**Pull request by msusanrbrown**

### CODE REVIEW

File: breathecode/services/slack/tests/commands/tests_student.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+from django.urls.base import reverse_lazy
+from rest_framework import status
+from ..mixins import SlackTestCase
+import breathecode.services.slack.commands as commands
+from ...exceptions import SlackException
+from ...commands.student import execute
+
+
+class SlackTestSuite(SlackTestCase):
+    def test_slack_command___users_is_an_empty_list(self):
```
Comment (jefer94) [inline breathecode/services/slack/tests/commands/tests_student.py:12]: Add tests about auth

File: breathecode/services/slack/tests/commands/tests_cohort.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+from unittest.mock import patch
+from rest_framework import status
+from ..mixins import SlackTestCase
+import breathecode.services.slack.commands as commands
+from ...exceptions import SlackException
+from ...commands.cohort import execute
+
+
+class SlackTestSuite(SlackTestCase):
+    def test_slack_command___cohort_does_not_exist(self):
```
Comment (jefer94) [inline breathecode/services/slack/tests/commands/tests_cohort.py:11]: Add tests about auth

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/services/slack/tests/client/tests_client.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    def test_slack_command___no_slack_user(self):
+        """Testing when user has no slack_user."""
+
+        data = {'text': 'student', 'user_id': 'name', 'team_id': 'team', 'channel_id': 'test'}
+        slack = Slack()
+        with self.assertRaisesMessage(SlackException, 'unauthorized-user'):
+            slack.execute_command(data)
+
+
+#    @patch('breathecode.services.slack.commands.madeup_name.execute', MagicMock(return_value=) )
```
Comment (jefer94) [inline breathecode/services/slack/tests/client/tests_client.py:30]: Add test passing the arguments to a fake command, make sure the arguments are being passed properly

File: breathecode/services/slack/tests/commands/tests_cohort.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                                        slack_user=slack_users,
+                                        capability='read_cohort',
+                                        user=1,
+                                        role='teacher',
+                                        academy=1,
+                                        slack_team=slack_team,
+                                        cohort_user=cohort_user,
+                                        slack_channel=slack_channel)
+
+        # teachers = CohortUser.objects.filter(cohort=cohort, role__in=['TEACHER', 'ASSISTANT'])
```
Comment (jefer94) [inline breathecode/services/slack/tests/commands/tests_cohort.py:120]: Remove this comment

File: breathecode/services/slack/tests/commands/tests_student.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        data = {
+            'text': 'student <@fdd2325|244372eew>',
+            'user_id': 'name',
+            'team_id': 'team',
+            'channel_id': 'test'
+        }
+        with self.assertRaisesMessage(SlackException, 'cohort-user-not-found'):
+            result = execute(users=['fdd2325'], context=data)
+
+    def test_slack_command___user_registered_in_a_cohort__without_financial_status_or_educational_status(
```
Comment (jefer94) [inline breathecode/services/slack/tests/commands/tests_student.py:54]: Add a case where the cohort user is register in two different cohorts

**Review by jefer94 — CHANGES_REQUESTED** [review]

**Comment by jefer94** [comment]
@msusanrbrown Fix those tests 
image

File: breathecode/services/slack/tests/commands/tests_student.py (hunk 5 above)
Comment (msusanrbrown) [reply breathecode/services/slack/tests/commands/tests_student.py:54]: it has been already implemented

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/feedback/views.py [hunk 6]
```diff
@@ -273,22 +274,23 @@ def delete(self, request, academy_id=None, survey_id=None):
...
-@permission_classes([AllowAny])
-def get_review_platform(request, platform_slug=None):
-
-    rp = ReviewPlatform.objects.all()
-    if platform_slug is not None:
-        rp = cu.filter(slug=platform_slug).first()
-        if rp is not None:
-            serializer = ReviewPlatformSerializer(items, many=False)
-            return Response(serializer.data)
+    @api_view(['GET'])
```
Comment (jefer94) [inline breathecode/feedback/views.py:277]: revert this file

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/services/slack/tests/commands/tests_student.py [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+                    'type': 'image',
+                    'image_url': '/static/img/avatar.png',
+                    'alt_text': f'{model.user[1].first_name} {model.user[1].last_name}',
+
+                }
+            }]
+        }
+
+        result = execute(users=['percybrown'], context=data)
+        print(result)
```
Comment (jefer94) [inline breathecode/services/slack/tests/commands/tests_student.py:172]: Remove print

File: breathecode/services/slack/commands/student.py [hunk 8]
```diff
@@ -14,22 +14,25 @@
...
 def execute(users, academies, **context):
 
     if len(users) == 0:
-        raise Exception('No usernames found on the command')
+        raise SlackException('No usernames found on the command', slug='users-not-provided')
 
     cohort_users = CohortUser.objects.filter(user__slackuser__slack_id=users[0],
                                              role='STUDENT',
                                              cohort__academy__id__in=[academies])
+    print(users[0])
```
Comment (jefer94) [inline breathecode/services/slack/commands/student.py:29]: Remove print

File: breathecode/services/slack/tests/client/tests_client.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+    def test_slack__execute_command__test_executor(self):
+        """Testing how execute_command is being executed."""
+
+        data = {'text': 'student', 'user_id': 'name', 'team_id': 'team', 'channel_id': 'test'}
+        slack = Slack()
+        expected = 'potato'
+        result = slack._execute_command(fake_command, 'fake', data)
+
+        print(fake_command.fake.execute.call_args_list)
```
Comment (jefer94) [inline breathecode/services/slack/tests/client/tests_client.py:63]: Remove print

**Review by jefer94 — CHANGES_REQUESTED** [review]

**Review by jefer94 — APPROVED** [review]

**Pull request merged**